### PR TITLE
Add new sddc datasource

### LINKED ISSUE #92: Feature Request :  Data Sources - vmc_sddc
Could consideration please be given to the extension of the provider to include a vmc_sddc data source ? I use the attributes of the SDDC objects in other modules/projects that consume the SDDCs and at the moment have to write a bunch of attributes (e.g. cloudadmin password, vsphere_uri, nsxt_rp_uri, UUID, etc.) to an external store (e.g. AWS Secrets Manager) and then using the aws_secretsmanager data object reading the values when consuming SDDCs in a different project when dynamically generating the objects.

This approach is OK however inefficient (using an external attribute store adds cost and duplicates a stale copy of the data) and makes code readability (passing hardcoded values into template files) when it would be nicer in most cases to just read and return the SDDC object using a data object.

Some example use cases:
* Avoid requirement when calling the vmware/nsxt provider in templates or other projects (for terraforming firewall rules for example) to hardcode host values (could use vmc_sddc datastore to make code cleaner)
* Avoid requirement when using hashicorp/vphere provider in  templates or other projects (for terraforming the vCenter) to use another provider (e.g. AWS Secrets Manager and/or DynamoDB) to store the cloudadmin pwd, user, vSphere URI etc.
* In Cloud Director service when terraforming the environment to be able to retrieve the values without using third party attribute store

Cheers

### PR DESCRIPTION
**Pull request by shylasrinivas**
Support for SDDC datasource

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ cd $GOPATH/src/github.com/terraform-providers
 git clone git@github.com:terraform-providers/terraform-provider-vmc.git
 ```
 
-After the clone has been completed, you can enter the provider directory and build the provider. 
+After the clone has been completed, you can enter the provider directory and build the provider.
```
Comment (rchadalawada) [inline README.md]: Simplify this sentence... After the clone is complete, ...

File: vmc/data_source_vmc_sddc.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"host_instance_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{HostInstancetypeI3, HostInstancetypeR5}, false),
```
Comment (rchadalawada) [inline vmc/data_source_vmc_sddc.go:131]: Are still supporting R5?

**Review by rchadalawada — COMMENTED** [review]
Looks OK to me

File: vmc/data_source_vmc_sddc.go (hunk 2 above)
Comment (shylasrinivas) [reply vmc/data_source_vmc_sddc.go:131]: Its the initial implementation that hasn't been modified. Why do you ask?

Comment (rchadalawada) [reply vmc/data_source_vmc_sddc.go:131]: I don’t think we support R5 anymore. Its irrelevant to this MR. So, no need to act on it in this MR. But, please check with VMC team.

**Review by rchadalawada — APPROVED** [review]

**Pull request closed without merging**